cfpb/cfgov-refresh
# Careers: unpublish JobListing pages if closed on USAJobs

USAJobs is the source of truth about the status of our job postings. When jobs close early on USAJobs (midnight Eastern), their pages are still live on our site until we manually unpublish them the next working day. This adds a script that can be automated to run after midnight to unpublish job pages if their related jobs have closed on USAJobs (ie, they have been moved to the USAJobs API archive endpoint, or their pages on USAJobs display a closed message instead of an apply button).

## Additions

- Script to unpublish live job pages if related job listings have closed on USAJobs


## Testing

1. Check out branch
2. Update USAJobs environment variables
3. Publish several expired job listing pages
4. Update database or add pages for [currently open jobs](https://www.consumerfinance.gov/about-us/careers/)
5. Run script and verify that pages for closed jobs have been unpublished and pages for open jobs remain live


## Notes

We currently only have access to jobs that are open to the public via the USAJobs archive endpoint, but that should be expanded to include internal/status jobs in an upcoming API release. In the meantime, this includes a temporary fallback check for closed text on the USAJobs job posting pages.


## Checklist
